llap-ext-client/src/java/org/apache/hadoop/hive/llap/LlapBaseInputFormat.java

```diff
@@ -157,12 +153,12 @@ public LlapBaseInputFormat() {
     HiveConf.setVar(job, HiveConf.ConfVars.LLAP_ZK_REGISTRY_USER, llapSplit.getLlapUser());
     SubmitWorkInfo submitWorkInfo = SubmitWorkInfo.fromBytes(llapSplit.getPlanBytes());
 
-    LlapServiceInstance serviceInstance = getServiceInstance(job, llapSplit);
-    String host = serviceInstance.getHost();
-    int llapSubmitPort = serviceInstance.getRpcPort();
+    final LlapDaemonInfo llapDaemonInfo = llapSplit.getLlapDaemonInfos()[0];
```

Comment:
can getLlapDaemonInfos() return empty array?

Reply:
no, it does not as of now. Same has been validated in get_splits. Added a comment about it in new patch.